mlprimitives/timeseries.py

```diff
@@ -0,0 +1,63 @@
+import pandas as pd
+import numpy as np
+
+
+def create_window_sequences(df_timeseries, window_size):
+    """
+        Function that takes in a Pandas.DataFrame and a window_size then creates
+            output arrays that correspond to a timeseries sequence with window_size overlap.
+            The output arrays can be fed into a timeseries forecasting model.
+        Inputs:
+            df_timeseries (Pandas.DataFrame): a Pandas dataframe which has 'timestamp'
+                and 'value' columns, and is sorted based on timestamp. 
+                The timestamp column is in UNIX format (in seconds).
+            window_size (int): number of values that overlap to create the sequence.
+        Outputs:
+            x (numpy.ndarray): contains the time series sequenced data.
+            y (numpy.ndarray): acts as the label for the forecasting problem.
+            time (numpy.ndarray): the corresponding timestamps series.
+    """
+    X = []
+    Y = []
+    time = []
+    for i in range(len(df_timeseries) - window_size):
+        X.append(df_timeseries[i: i + window_size]['value'].values.copy().reshape([-1, 1]))
+        Y.append(df_timeseries[i + 1: i + window_size + 1]['value'].values.copy().reshape([-1, 1]))
+        time.append(df_timeseries.iloc[i + window_size]['timestamp'])
+    return np.asarray(X), np.asarray(Y), np.asarray(time)
+
+
+def aggregate_average_time(df_time_value, interval_time_delta, start_time, end_time):
+    """
+        Function that aggregates data in a Pandas dataframe by averaging over a given interval. 
+            It starts averaging from specified start_time.
+        Inputs:
+            df_time_value (Pandas.DataFrame): a Pandas dataframe which has 'timestamp' 
+                and 'value' columns, and is sorted based on timestamp. The timestamp
+                column is in UNIX format (in seconds).
+            interval_time_delta (int): an Integer denoting the number of seconds 
+                in the desired interval.
+            start_time (int): a UNIX time stamp indicating the time to start
+                aggregating. Can be smaller than the smallest time stamp value in the dataframe.
```

Comment:
> Can be smaller than the smallest time stamp value in the dataframe.
Is this right? If I'm reading it correctly, the current implementation would generate empty rows in the dataframe if this happens.
The same happens with the end_date.

Reply:
Yes, because if you want to do a multi-PID (multi-signal) LSTM then you are not sure that all the starting times would line up. Technically, if the default value for `start_time` is the smallest timestamp in the dataframe, then we can remove `start_time` and find the intervals starting from the smallest time. The same applies to `end_time` but the problem is, if we are using multi-signal LSTM then we would need to make sure the timestamps match up. What do you think?

Reply:
That's true. But right now this should be out of the scope of this primitive.
Let's focus on a single table and make the default values be the min and the max. If someone wants to use this in more than one signal at a time, it's their responsibility to make sure that they are aligned.